### ISSUE TITLE
Bug on my modpack, the pack crash during start (Exception: java.lang.OutOfMemoryError)

### ISSUE THREAD

**Issue by Fireztonez** [codetaylor/dropt#23]
Hello,
After several updates of mods, it is impossible to start the pack with your mod. As soon as I activate your mod, my modpack crash with an error;
"Exception: java.lang.OutOfMemoryError"
(I allow more than 9go of Ram now, so for sure is not realy a problem with not enough memory allow to java... The pack run great with less than that without Dropt...)

I know that it is very probably a conflict with another mod, but I can not find out what mod it is. So I send you the boot logs, possibly that you will be able to see where the problem is coming from.

Tell me if you are ever able to give me a track on the problem or if you can do some things for that. Otherwise, I will continue to try to find where the error comes from adding one mod at a time (I already spent about 3 hours to do this, without finding for the moment) ...

https://gist.github.com/Fireztonez/51f7a9065f5f777683ab75c335bed1c7

Minecraft 1.12.2
Dropt 1.12.2 - 1.8.2
Forge 14.23.2.2632
Java 1.8.0_161

**Comment by codetaylor** [comment]
Yeah, I really have no idea. Let me know if you get more information.

**Comment by Fireztonez** [comment]
Ok, no problem!

**Comment by Fireztonez** [comment]
I found the problem!

The crash append when I active the option : 
`    # Set to true to inject about 100,000 unique rules to assist in profiling.
    # Used in development to test performance.
    B:INJECT_PROFILING_RULES=false
`

**Comment by codetaylor** [comment]
Yeah, that isn't meant to be used in a modded environment. In an environment with just Vanilla and Dropt, it will add about 100,000 unique rules. It does so by creating a rule for every block / item combination. So if you have 100 block and 1000 items, that's 100 * 1000, or 100,000 rules. In a modded environment the number of blocks and items can be much greater than in Vanilla and the resulting ruleset will be exponentially greater.

TLDR; don't enable that option in a modpack! :)

**Comment by codetaylor** [comment]
@Fireztonez Also make sure that `"debug": true` isn't enabled on any rules. That flag should only be selectively enabled on one to a few recipes at a time and promptly disabled because the log file will get massive. You wouldn't want to leave those flags enabled and distribute the modpack to other people... :)

**Comment by Fireztonez** [comment]
Ok, thank's for that information,
I know, I will not distribute with the debug at true, my pack is in early developpement stade now, but thank's for the warning about the fact of getting a really massive files logs!!

Fireztonez
